### PR TITLE
[CWS] llvm compiler improvements

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR cleans up a bit the llvm runtime compilation helper.
Basically it:
- removes pointers when they are not needed
- replaces usage of map with an array
- removes unneeded link options
- general cleanup

This code is still not perfect, very far from it, it's just a tiny bit better. The end-goal is still https://github.com/DataDog/datadog-agent/pull/10772

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/ebpf/compiler/compiler.cpp [hunk 1]
```diff
@@ -11,63 +11,68 @@
...
         errString = "could not lookup target";
         return;
     }
 
     llvm::TargetOptions targetOptions;
-    auto RM = llvm::Optional<llvm::Reloc::Model>();
-    targetMachine = std::unique_ptr<llvm::TargetMachine>(theTarget->createTargetMachine(
-        theTriple.getTriple(), "generic", "", targetOptions, RM, llvm::None, llvm::CodeGenOpt::Aggressive));
+    targetMachine = std::unique_ptr<llvm::TargetMachine>(llvmTarget->createTargetMachine(
+        targetTriple.getTriple(), "generic", "", targetOptions, {}, llvm::None, llvm::CodeGenOpt::Aggressive));
```
Comment (brycekahle) [inline pkg/ebpf/compiler/compiler.cpp:75]: Do you know the significance of removing the `RM` argument here?

**Comment by brycekahle** [comment]
Are there any runtime benefits to this change? I'm hesitant to change something so late in the release cycle, especially when we are likely going to be replacing it in the next release.

File: pkg/ebpf/compiler/compiler.cpp (hunk 1 above)
Comment (paulcacheux) [reply pkg/ebpf/compiler/compiler.cpp:75]: `RM` was defined as `auto RM = llvm::Optional<llvm::Reloc::Model>();` so basically the empty optional variant. So the semantic is preserved here

**Comment by paulcacheux** [comment]
It's a small improvement, I completely understand if we don't want to merge it. Maybe we can keep it on the side, and if we again have issues with runtime compilation use it then to test if it helps

**Pull request closed without merging**